Zilon.Core/Zilon.Core/World/IGlobeInitializer.cs

```diff
@@ -0,0 +1,472 @@
+            foreach (var actorDataItem in actorDataListMaterialized)
+            {
+                var actor = actorDataItem.Actor;
+                var sector = actorDataItem.Sector;
+
+                var taskSource = actor.TaskSource;
+
+                var context = new SectorTaskSourceContext(sector);
+
+                //TODO Это можно делать параллельно. Одновременно должны думать сразу все актёры.
+                // Делается легко, через Parallel.ForEach, но из-за этого часто заваливаются тесты и даже не видно причины отказа.
+                var actorTask = await taskSource.GetActorTaskAsync(actor, context).ConfigureAwait(false);
+
+                var state = new TaskState(actor, actorTask, taskSource);
+                if (!_taskDict.TryAdd(actor, state))
+                {
+                    // Это происходит, когда игрок пытается присвоить новую команду,
+                    // когда старая еще не закончена и не может быть заменена.
+                    throw new InvalidOperationException("Попытка назначить задачу, когда старая еще не удалена.");
+                }
+            }
+        }
+
+        private static void ProcessTasks(IDictionary<IActor, TaskState> taskDict)
+        {
+            var states = taskDict.Values;
+            // Все актёры еще раз сортируются, чтобы зафиксировать поведение для тестов.
+            // Может быть ситуация, когда найдено одновременно два актёра без задач (в самом начале теста, например).
+            // В этом случае порядок элементов, полученных из states, не фиксирован.
+            // В тестах мжет быть важно, в каком порядке ходят актёры, чтобы корректно обрабатывать коллизии.
+            // Поэтому тут выполняется еще одна сортировка, от которой, по-хорошему нужно избавиться.
+            // Лучше сразу иметь список states с сортировкой по приоритету при добавлении элемента.
+            var orderedStates = states.OrderBy(x => x.Actor.Person.Id);
+            foreach (var state in orderedStates)
+            {
+                if (!state.Actor.CanExecuteTasks)
+                {
+                    // По сути, возможность выполнять задачи - это основая роль этого объекта.
+                    // песонаж может перестать выполнять задачи по следующим причинам:
+                    // 1. Персонажи, у которых есть модуль выживания, могут умереть, пока выполняют текущую задачу (выполняются предусловия).
+                    // Мертвые персонажы не выполняют задач.
+                    // Их задачи игнорируем, т.к. задачи могут выполниться после смерти персонажа,
+                    // что противоречит логике задач.
+                    continue;
+                }
+
+                state.UpdateCounter();
+
+                if (state.TaskIsExecuting)
+                {
+                    try
+                    {
+                        state.Task.Execute();
+                    }
+                    catch (Exception exception)
```

Comment:
Codacy found an issue: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=6105716)